remote/slack/helper.go

```diff
@@ -474,56 +476,54 @@ func readFromEventsAPI(api *slack.Client, vToken string, inputMsgs chan<- models
-						bot.Log.Errorf("Did not get Slack user info: %s", err.Error())
-					}
-					timestamp := ev.Timestamp
-					threadTimestamp := ev.ThreadTimestamp
-					inputMsgs <- populateMessage(models.NewMessage(), msgType, channel, text, timestamp, threadTimestamp, mentioned, user, bot)
-				}
-			case *slack.ConnectedEvent:
-				// populate users
-				populateBotUsers(ev.Info.Users, bot)
-				// populate user groups
-				populateUserGroups(bot)
-				bot.Log.Debugf("RTM connection established!")
-			case *slack.GroupJoinedEvent:
-				// when the bot joins a channel add it to the internal lookup
-				// NOTE: looks like there is another unsupported event we could use
-				//   Received unmapped event \"member_joined_channel\"
-				// Maybe watch ffor an update to slack package for future support
-				if bot.Rooms[ev.Channel.Name] == "" {
-					bot.Rooms[ev.Channel.Name] = ev.Channel.ID
-					bot.Log.Debugf("Joined new channel. %s(%s) added to lookup", ev.Channel.Name, ev.Channel.ID)
+		msg := <-rtm.IncomingEvents
+		switch ev := msg.Data.(type) {
+		case *slack.MessageEvent:
+			senderID := ev.User
+			// Sometimes message events in RTM don't have a User ID?
+			// Also, only process messages that aren't from the bot itself
+			if senderID != "" && bot.ID != senderID {
+				channel := ev.Channel
+				msgType, err := getMessageType(channel)
+				if err != nil {
+					bot.Log.Debug(err.Error())
 				}
-			case *slack.HelloEvent:
-				// ignore - this is the very first initial event sent when connecting to Slack
-			case *slack.RTMError:
-				bot.Log.Error(ev.Error())
-			case *slack.ConnectionErrorEvent:
-				bot.Log.Errorf("RTM connection error: %+v", ev)
-			case *slack.InvalidAuthEvent:
-				if !bot.CLI {
-					bot.Log.Debug("Invalid Authorization. Please double check your Slack token.")
+				text, mentioned := removeBotMention(ev.Text, bot.ID)
+				user, err := rtm.GetUserInfo(senderID)
+				if err != nil && senderID != "" { // we only care if senderID is not empty and there's an error (senderID == "" could be a thread from a message)
+					bot.Log.Errorf("Did not get Slack user info: %s", err.Error())
 				}
-			} // EOF inner switch
-		} // EOF outter switch
-	} // EOF for
+				timestamp := ev.Timestamp
+				threadTimestamp := ev.ThreadTimestamp
+				inputMsgs <- populateMessage(models.NewMessage(), msgType, channel, text, timestamp, threadTimestamp, mentioned, user, bot)
+			}
+		case *slack.ConnectedEvent:
+			// populate users
+			populateBotUsers(ev.Info.Users, bot)
+			// populate user groups
+			populateUserGroups(bot)
+			bot.Log.Debugf("RTM connection established!")
+		case *slack.GroupJoinedEvent:
+			// when the bot joins a channel add it to the internal lookup
+			// NOTE: looks like there is another unsupported event we could use
+			//   Received unmapped event \"member_joined_channel\"
+			// Maybe watch ffor an update to slack package for future support
```

Comment:
Should be for instead of ffor. Also, it looks like this event has been added to the slack library we are using - https://github.com/nlopes/slack/search?utf8=%E2%9C%93&q=MemberJoinedChannelEvent&type=.

Reply:
Thanks. I will have to test and see if that event makes sense.

Reply:
Adding support for that new event will also help to resolve https://github.com/target/flottbot/issues/99.

Reply:
we're doing something funky in this area.. will make that new event and cleanup part of another PR